### PR TITLE
[6.x] Game/Unit: Redefine Shapeshift Forms from 6.x DBC

### PR DESCRIPTION
**Pull request by P-Kito**
**Changes proposed**:
- Define DBC Shapeshift Forms for 6.x

**Target branch(es)**: 6x

### CODE REVIEW

**Comment by Shauren** [comment]
I have a request - please fill them from SpellShapeshiftForm.dbc (and rename existing ones)
Expected result should look like this
`FORM_(dbcname, replacing spaces and other special characters with underscore, then replacing all multiple underscores with a single one -> a - qq -> a___qq -> a_qq) = decimal_value (not hex)`

**Comment by P-Kito** [comment]
For cat would mean 2x Form in name:  FORM_CAT_FORM  (dbc: Cat Form)

**Comment by Shauren** [comment]
i see only one named Cat Form there, the other one is "Creature - Cat" FORM_CREATURE_CAT

**Comment by P-Kito** [comment]
Travis will tell me left-overs, but I think I covered them all.

File: src/server/game/Entities/Unit/Unit.h [hunk 1]
```diff
@@ -260,37 +260,40 @@ enum UnitBytes1_Flags
...
+    FORM_BERSERKER_STANCE           = 19,
+    FORM_SERPENT_STANCE             = 20,
+    FORM_ZOMBIE                     = 21,
+    FORM_METAMORPHOSIS              = 22,
+    FORM_OX_STANCE                  = 23,
+    FORM_TIGER_STANCE               = 24,
+    FORM_UNDEAD                     = 25,
+    FORM_FRENZY                     = 26,
+    FORM_FLIGHT_FORM_EPIC           = 27,
+    FORM_SHADOW_FORM                = 28,
```
Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.h]: almost, almost perfect - FORM_SHADOWFORM

File: src/server/game/Entities/Unit/Unit.h [hunk 2]
```diff
@@ -260,37 +260,40 @@ enum UnitBytes1_Flags
...
+    FORM_BERSERKER_STANCE           = 19,
+    FORM_SERPENT_STANCE             = 20,
+    FORM_ZOMBIE                     = 21,
+    FORM_METAMORPHOSIS              = 22,
+    FORM_OX_STANCE                  = 23,
+    FORM_TIGER_STANCE               = 24,
+    FORM_UNDEAD                     = 25,
+    FORM_FRENZY                     = 26,
+    FORM_FLIGHT_FORM_EPIC           = 27,
+    SPELL_SHADOWFORM                = 28,
```
Comment (Shauren) [inline src/server/game/Entities/Unit/Unit.h]: i think this is a bigger derp than before :D

Comment (P-Kito) [inline src/server/game/Entities/Unit/Unit.h]: ...............................

**Pull request merged**